### PR TITLE
Use toolchain action v2

### LINKED ISSUE #1430: Github actions-rs/toolchain is no longer supported
Need to change to a supported GH Action for the toolchain.  https://github.com/dtolnay/rust-toolchain

### LINKED ISSUE #1438: Github Action actions-rs/tarpaulin is no longer supported
Need to replace with [xd009642/tarpaulin](https://github.com/xd009642/tarpaulin) which is currently being maintained.

### LINKED ISSUE #1469: Remove hard coding of version number in GH Actions
Read the version number from the Cargo.toml and export it as an environment variable that can be used in the rest of the GH Actions.

### LINKED ISSUE #1470: rust-lint and rust-lint-skipped should have the same paths and steps
Inconsistent paths and steps are causing the GitHub Checks to fail.

### LINKED ISSUE #1471: Protoc Setup Action should use Node 16
Node 12 is being removed in the near future.  Need to use Node 16 instead in the action.

### PR DESCRIPTION
**Pull request by sbtaylor15**
## Description

Use the Pyrsia rust-toolchain v2 action that removes the dependency on actions-rs actions that are no longer supported.  Also, uses Node 16 for the protoc install.  Finally, exposes the Package Version as PKG_VERSION env var to simplify the workflow. 

Fixes #1471
Fixes #1470
Fixes #1469
Fixes #1438
Fixes #1430

## PR Checklist

- [X] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [X] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/community/get_involved/good_pr.md)
- [X] I've included a good title and brief description along with how to review them.
- [X] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [X] I've run [pre-commit.sh](https://github.com/pyrsia/pyrsia/blob/main/pre-commit.sh) ([pre-commit.bat](https://github.com/pyrsia/pyrsia/blob/main/pre-commit.bat)) successfully.
- [X] I've run the [integrations tests](https://github.com/pyrsia/pyrsia-integration-tests#how-to-set-up-and-run-the-tests)
on the PR branch and all tests passes.

### CODE REVIEW

**Review by karolh2000 — APPROVED** [review]

**Review by dasmanas — COMMENTED** [review]
Should we consider to be definitive while choosing shell. Reference [jobs.<job_id>.steps[*].shell
](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell)

File: .github/workflows/rust-lint.yml [hunk 1]
```diff
@@ -14,34 +14,32 @@ jobs:
...
+          packages: cargo-audit   
+      - run: |
+          cargo audit
 
   rustfmt:
     runs-on: ubuntu-20.04
     steps:
       - uses: actions/checkout@v3
-      - uses: pyrsia/rust-toolchain@v1
+      - uses: pyrsia/rust-toolchain@v2
```
Comment (fishseabowl) [inline .github/workflows/rust-lint.yml:27]:
```suggestion
      - uses: pyrsia/rust-toolchain@v2
```

**Review by fishseabowl — DISMISSED** [review]
lgtm

**Comment by sbtaylor15** [comment]
Linted the yaml files and fixed the common errors per @fishseabowl suggestion.  This added a few more files to the PR.

**Review by dasmanas — APPROVED** [review]
lgtm

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -268,7 +268,7 @@ jobs:
 
       - name: Run cargo-tarpaulin
         run: |
-          cargo tarpaulin --workspace --lib --bins --tests --benches --examples --out Lcov -- --test-threads 1
+          cargo tarpaulin --workspace --lib --bins --tests --benches --examples --out Lcov --output-dir coverage -- --test-threads 1
```
[the remark this answers is not in the document]
Comment (sbtaylor15) [reply .github/workflows/rust.yml:271]: @fishseabowl some tests break if they are executed out of order.  I believe some of the tests use the same directory for files.  So one test may delete a file that another one is trying to use that same file.

[the remark this answers is not in the document]
Comment (dasmanas) [reply .github/workflows/rust.yml:271]: `--test-threads 1` makes sure test runs in single thread. In cases where test might hit the same resource and create inconsistency.

**Comment by sbtaylor15** [comment]
closing PR and opening a new to ensure coveralls.io is working as expected.

**Pull request closed without merging**